src/mp_api/client.py

```diff
@@ -390,12 +401,293 @@ def get_entries(
+
+        # download the ion reference data from MPContribs
+        ion_data = self.get_ion_reference_data(chemsys)
+
+        # build the PhaseDiagram for get_ion_entries
+        ion_ref_comps = [Ion.from_formula(d["formula"]).composition for d in ion_data]
+        ion_ref_elts = list(
+            itertools.chain.from_iterable(i.elements for i in ion_ref_comps)
+        )
+        ion_ref_entries = self.get_entries_in_chemsys(
+            list(set([str(e) for e in ion_ref_elts] + ["O", "H"])),
+        )
+
+        # suppress the warning about supplying the required energies; they will be calculated from the
+        # entries we get from MPRester
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                "ignore",
+                message="You did not provide the required O2 and H2O energies.",
+            )
+            compat = MaterialsProjectAqueousCompatibility(solid_compat=solid_compat)
+        # suppress the warning about missing oxidation states
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                "ignore", message="Failed to guess oxidation states.*"
+            )
+            ion_ref_entries = compat.process_entries(ion_ref_entries)
+        if use_gibbs:
+            # replace the entries with GibbsComputedStructureEntry
+            from pymatgen.entries.computed_entries import GibbsComputedStructureEntry
+
+            ion_ref_entries = GibbsComputedStructureEntry.from_entries(
+                ion_ref_entries, temp=300
+            )
+        ion_ref_pd = PhaseDiagram(ion_ref_entries)
+
+        ion_entries = self.get_ion_entries(ion_ref_pd, ion_ref_data=ion_data)
+        pbx_entries = [PourbaixEntry(e, f"ion-{n}") for n, e in enumerate(ion_entries)]
+
+        # Construct the solid pourbaix entries from filtered ion_ref entries
+        ion_ref_comps = [e.composition for e in ion_entries]
+        ion_ref_elts = list(
+            itertools.chain.from_iterable(i.elements for i in ion_ref_comps)
+        )
+        extra_elts = (
+            set(ion_ref_elts)
+            - {Element(s) for s in chemsys}
+            - {Element("H"), Element("O")}
+        )
+        for entry in ion_ref_entries:
+            entry_elts = set(entry.composition.elements)
+            # Ensure no OH chemsys or extraneous elements from ion references
+            if not (
+                entry_elts <= {Element("H"), Element("O")}
+                or extra_elts.intersection(entry_elts)
+            ):
+                # Create new computed entry
+                form_e = ion_ref_pd.get_form_energy(entry)
+                new_entry = ComputedEntry(
+                    entry.composition, form_e, entry_id=entry.entry_id
+                )
+                pbx_entry = PourbaixEntry(new_entry)
+                pbx_entries.append(pbx_entry)
+
+        return pbx_entries
+
+    # TODO - @lru_cache causes this method to fail when chemsys is given as a list,
```

Comment:
You could use `Union[str, Tuple]` instead, which is hashable.

Reply:
I'd prefer to keep the call signature consistent with `get_pourbaix_entries` and other methods.

Reply:
To keep it consistent with the legacy method? It _has_ to be tuple/hashable if you want to use @lru_cache, it cannot be list. Change is worth it imo to make the function cachable.

Reply:
to keep consistent with `get_entries` or `get_entries_in_chemsys`. Users expect to be able to specify a chemsys as either a string `Li-Fe-O` or a list `['Li','Fe','O']`, right? I don't recall seeing anywhere where we specify the chemsys as a tuple

Reply:
Right, but we can change that. Or you can make this a private method that accepts a tuple and the public method accepts a list and converts it. Either way. But I think it's an important method to cache.

Reply:
Sure, no argument at all with that approach; I just want to keep the user-facing arguments consistent